exense-commons-parent/exense-basic-commons/src/main/java/ch/exense/commons/processes/ExternalJVMLauncher.java

```diff
@@ -16,59 +16,97 @@
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.nio.charset.StandardCharsets;
 import java.util.ArrayList;
 import java.util.List;
 
+import ch.exense.commons.io.FileHelper;
 import ch.exense.commons.processes.ManagedProcess.ManagedProcessException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class ExternalJVMLauncher {
 
+	private static final Logger logger = LoggerFactory.getLogger(ExternalJVMLauncher.class);
 	private final String javaPath;
 	
 	private final File processLogFolder;
-	
+
 	public ExternalJVMLauncher(String javaPath, File processLogFolder) {
 		super();
 		this.javaPath = javaPath;
 		this.processLogFolder = processLogFolder;
 	}
 
-	private String buildClasspath() {
+	private File buildClasspath() throws IOException {
 		//Fix for Java11 compatibility (Classloader is no more an instance of URLClassLoader)
 		String[] classPathEntries = System.getProperty("java.class.path").split(File.pathSeparator);
 
-		StringBuilder cp = new StringBuilder();
-		cp.append("\"");
-		for(String path:classPathEntries) {
-			cp.append(new File(path).getAbsolutePath()+File.pathSeparator);
+		File javaClassPathArgsFile = FileHelper.createTempFile();
+
+		try (OutputStreamWriter cp = new OutputStreamWriter(new FileOutputStream(javaClassPathArgsFile), StandardCharsets.UTF_8)) {
+			cp.append("-cp ");
+			cp.append("\"");
+			for(String path:classPathEntries) {
+				String absolutePath = new File(path).getCanonicalPath();
+				absolutePath = (isWindows()) ? absolutePath.replace("\\","/") : absolutePath;
+				cp.append(absolutePath);
+				cp.append(File.pathSeparator);
+			}
+			cp.append("\"");
 		}
-		cp.append("\"");
-		return cp.toString();
+		return javaClassPathArgsFile;
 	}
 
 	public ManagedProcess launchExternalJVM(String name, Class<?> mainClass, List<String> vmargs, List<String> progargs) throws ManagedProcessException {
 		return launchExternalJVM(name, mainClass, vmargs, progargs, true);
 	}
 	
 	public ManagedProcess launchExternalJVM(String name, Class<?> mainClass, List<String> vmargs, List<String> progargs, boolean redirectOutput) throws ManagedProcessException {
-		String cp = buildClasspath();
-		
-		List<String> cmd = new ArrayList<>();
-		cmd.add(javaPath);
-		cmd.add("-cp");
-		cmd.add(cp);
-		
-		cmd.addAll(vmargs);
-		
-		cmd.add(mainClass.getName());
-		
-		cmd.addAll(progargs);
-		
-		ManagedProcess process = new ManagedProcess(name, cmd, processLogFolder, redirectOutput);
-		process.start();
-		return process;
+		try {
+			File cpArgFile = buildClasspath();
+			List<String> cmd = new ArrayList<>();
+			cmd.add(javaPath);
+			cmd.add("@" + cpArgFile.getCanonicalPath());
+
+			cmd.addAll(vmargs);
+
+			cmd.add(mainClass.getName());
+
+			cmd.addAll(progargs);
+
+			ManagedProcess process = new JvmManagedProcess(cpArgFile, name, cmd, processLogFolder, redirectOutput);
+			process.start();
+			return process;
+		} catch (IOException e) {
```

Comment:
Thie catch seems to be related to the method buildClasspath only. I would throw the ManagedProcessException directly in buildClasspath